### PR TITLE
Add AudioMeter widget and refactor audio silence detection

### PR DESCRIPTION
**Pull request by ItsSimko**
Issues: #21 #13 #22 

## Summary by Sourcery

Add an AudioMeter widget to the settings UI for adjusting audio silence cutoff thresholds and refactor audio processing to use this configurable threshold. Update the build and CI configurations to accommodate these changes.

New Features:
- Introduce an AudioMeter widget to the settings UI, allowing users to visually adjust and monitor audio silence cutoff thresholds.

Enhancements:
- Refactor the audio silence detection logic to use a configurable silence cutoff threshold, improving flexibility in audio processing.

Build:
- Update the NSIS installer script to include additional files in the installation package.

CI:
- Modify the release workflow to remove the '--onefile' option from the PyInstaller command.

### CODE REVIEW

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: src/FreeScribe.client/client.py [hunk 1]
```diff
@@ -141,48 +139,52 @@ def toggle_pause():
...
 
-def is_silent(data, threshold=float(app_settings.editable_settings["Silence cut-off"])):
-    max_value = max(data)
-    #print(f"Max audio value: {max_value}")
+def is_silent(data, threshold=0.01):
+    """Check if audio chunk is silent"""
+    max_value = max(abs(np.array(data)))
     return max_value < threshold
 
 def realtime_text():
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:181]: https://github.com/ClinicianFOCUS/FreeScribe/pull/23/commits/2f42662d5d8c795e7d08ee93e5850d31daac66cb

File: src/FreeScribe.client/client.py [hunk 2]
```diff
@@ -141,48 +139,52 @@ def toggle_pause():
 
 def record_audio():
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:140]: Could be done but this is old code so i left it as is... maybe refactor one day?

File: src/FreeScribe.client/client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:181]: https://github.com/ClinicianFOCUS/FreeScribe/pull/23/commits/2f42662d5d8c795e7d08ee93e5850d31daac66cb

File: src/FreeScribe.client/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py:140]: Could be refactored... but this is from the old code... so i left it as is and made the changes necessary. Could be done if we choose to refactor all of the code?

File: src/FreeScribe.client/client.py [hunk 3]
```diff
@@ -141,48 +139,52 @@ def toggle_pause():
...
-    global frames, is_realtimeactive
+    global frames, is_realtimeactive, audio_queue
     if not is_realtimeactive:
         is_realtimeactive = True
-        model_name = app_settings.editable_settings["Whisper Model"].strip()
-        model = whisper.load_model(model_name)
+        model = None
+        if app_settings.editable_settings["Real Time"]:
+            model_name = app_settings.editable_settings["Whisper Model"].strip()
+            model = whisper.load_model(model_name)
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/client.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/23/commits/4a1d5854b85e442dacb5e087c9ccc207dacdbb9c

File: src/FreeScribe.client/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yingbull) [reply src/FreeScribe.client/client.py:140]: If it works I am okay with it for now but yes this is ideal; where we can let's do this as it just ends up as debt in the long term but agree some things are larger than we can do on this assigned effort.

**Review by yingbull — APPROVED** [review]

**Pull request merged**